climpred/bootstrap.py

```diff
@@ -173,81 +212,133 @@ def bootstrap_perfect_model(ds,
-        pers = xr.concat(pers, dim='bootstrap')
-
-    def _distribution_to_ci(ds, ci_low, ci_high, dim='bootstrap'):
-        try:
-            if len(ds.chunks) >= 1:
-                ds = ds.compute()
-        except:
-            pass
-        ds_ci = ds.quantile(q=[ci_low, ci_high], dim=dim)
-        return ds_ci
-
-    if compute_ci:
-        init_ci = _distribution_to_ci(init, ci_low, ci_high)
-        if compute_uninitialized_skill:
-            uninit_ci = _distribution_to_ci(uninit, ci_low, ci_high)
-        if compute_persistence_skill:
-            pers_ci = _distribution_to_ci(pers, ci_low_pers, ci_high_pers)
-
-    else:
-        init_ci = None
-        pers_ci = None
-        uninit_ci = None
-
-    def _pvalue_from_distributions(simple_fct, init, metric=metric):
-        """Get probability that skill of simple_fct is larger than
-        init skill."""
-        pv = ((simple_fct - init) > 0).sum('bootstrap') / init.bootstrap.size
-        if metric not in ['pearson_r', 'ppp']:  # positively oriented metrics
-            pv = 1 - pv
-        return pv
-
-    if compute_uninitialized_skill:
-        p_uninit_over_init = _pvalue_from_distributions(uninit, init)
-    else:
-        p_uninit_over_init, uninit_ci = None, None
-
-    if compute_persistence_skill:
-        p_pers_over_init = _pvalue_from_distributions(pers, init)
-
-    else:
-        p_pers_over_init, pers_ci = None, None
+    uninit = xr.concat(uninit, dim='bootstrap')
+    pers = xr.concat(pers, dim='bootstrap')
+
+    init_ci = _distribution_to_ci(init, ci_low, ci_high)
+    uninit_ci = _distribution_to_ci(uninit, ci_low, ci_high)
+    pers_ci = _distribution_to_ci(pers, ci_low_pers, ci_high_pers)
+
+    p_uninit_over_init = _pvalue_from_distributions(uninit, init)
+    p_pers_over_init = _pvalue_from_distributions(pers, init)
 
     # calc skill
     init_skill = compute_perfect_model(
-        ds, control, metric=metric, comparison=comparison, running=running,
-        reference_period=reference_period)
+        ds, metric=metric, comparison=comparison)
+    uninit_skill = uninit.mean('bootstrap')
+    pers_skill = compute_persistence(
+        ds, control, metric=metric)
+
+    # somehow there may be a member dim, which lets concat crash
+    if 'member' in init_skill:
+        del init_skill['member']
+    # wrap results together in one dataarray
+    skill = xr.concat([init_skill.squeeze(), uninit_skill, pers_skill], 'i')
+    skill['i'] = ['init', 'uninit', 'pers']
+
+    # probability that i beats init
+    p = xr.concat([p_uninit_over_init, p_pers_over_init], 'i')
+    p['i'] = ['uninit', 'pers']
+
+    # ci for each skill
+    ci = xr.concat([init_ci, uninit_ci, pers_ci],
+                   'i').rename({'quantile': 'results'})
+    ci['i'] = ['init', 'uninit', 'pers']
+
+    results = xr.concat([skill, p], 'results')
+    results['results'] = ['skill', 'p']
+    # (RXB) Drop any spurious coordinates that came along with results due to
```

Comment:
thanks for those remarks. decorator for this would be cool I guess or at least a util function.

Reply:
Yep this should be a v2 thing. But there's frequently issues with merging datasets, running `xskillscore` metrics, etc. when coordinates don't match exactly or when there's not a coordinate to go with the dimension passed to `xskillscore`.